inc/class-plugin.php

```diff
@@ -157,36 +227,50 @@ public function get_s3_url() {
 	 *
 	 * @return string
 	 */
-	public function get_s3_bucket() {
-		return $bucket = strtok( $this->bucket, '/' );
+	public function get_s3_bucket() : string {
+		return strtok( $this->bucket, '/' );
 	}
 
-	public function get_s3_bucket_region() {
+	/**
+	 * Get the region of the S3 bucket.
+	 *
+	 * @return string
+	 */
+	public function get_s3_bucket_region() : ?string {
 		return $this->region;
 	}
 
-	public function get_original_upload_dir() {
+	/**
+	 * Get the original upload directory before it was replaced by S3 uploads.
+	 *
+	 * @return array{path: string, basedir: string, baseurl: string, url: string}
+	 */
+	public function get_original_upload_dir() : array {
 
 		if ( empty( $this->original_upload_dir ) ) {
 			wp_upload_dir();
 		}
 
-		return $this->original_upload_dir;
+		/**
+		 * @var array{path: string, basedir: string, baseurl: string, url: string}
+		 */
+		$upload_dir = $this->original_upload_dir;
+		return $upload_dir;
```

Comment:
Why is this a separate var?

Reply:
Due to Psalm's comment parsing above an assign expression rather than return.